### PR TITLE
chore: remove `extend` dependency

### PR DESCRIPTION
**Pull request by gameroman**
##### Checklist

- [ ] `npm test` passes (tip: `npm run lint-fix` can correct most style issues)
- [ ] tests are included
- [ ] documentation is changed or added

### CODE REVIEW

**Comment by pyrooka** [comment]
Hey @gameroman, thanks for the contribution. We'll try to look into the PR soon, but I'd like to ask two things in the meantime:
1. Sign off all your commits in the branch, otherwise the DCO check will keep failing
2. Since the purpose of this PR is to remove the `extend` dependency from the project, it should be also removed from the package.json - along with package-lock.json -, right?

File: lib/cookie-support.ts [hunk 1]
```diff
@@ -34,8 +33,7 @@ const internalCreateCookieInterceptor = (cookieJar: CookieJar) => {
       const cookieHeaderValue = await cookieJar.getCookieString(config.url);
       if (cookieHeaderValue) {
         logger.debug('CookieInterceptor: setting cookie header');
-        const cookieHeader = { cookie: cookieHeaderValue };
-        config.headers = extend(true, {}, config.headers, cookieHeader);
+        config.headers.set('cookie', cookieHeaderValue);
```
Comment (diatrcz) [inline lib/cookie-support.ts]:
```suggestion
        config.headers['cookie'] = cookieHeaderValue;
```

File: lib/request-wrapper.ts [hunk 2]
```diff
@@ -248,9 +245,11 @@ export class RequestWrapper {
...
    */
   public async sendRequest(parameters): Promise<any> {
-    const options = extend(true, {}, parameters.defaultOptions, parameters.options);
-    const { path, body, form, formData, qs, method, serviceUrl, axiosOptions } = options;
-    let { headers, url } = options;
+    const options = { ...parameters.defaultOptions, ...parameters.options };
+    let headers = { ...parameters.defaultOptions?.headers, ...parameters.options?.headers };
+    const qs = { ...parameters.defaultOptions?.qs, ...parameters.options?.qs };
+    const { path, body, form, formData, method, serviceUrl, axiosOptions } = options;
+    let { url } = options;
```
Comment (diatrcz) [inline lib/request-wrapper.ts]: With this change if `parameters.options` contains nested objects, they get replaced entirely rather than merged with. This differs from the previous deep merge behaviour provided by `extend`.

I added a new test case in the main branch testing this specific edge case.

**Review by diatrcz — CHANGES_REQUESTED** [review]

**Comment by diatrcz** [comment]
Hi @gameroman, thank you for the contribution. Additionally to the change requests:

1. As stated above, please sign your commits so DCO stops failing and remove `extend` from `package.json` and `package-lock.json`.
2. Please keep the branch up to date with the main branch.

**Comment by gameroman** [comment]
Thank you for the review, I’ll fix everything when I get back👍

File: lib/request-wrapper.ts (hunk 2 above)
Comment (diatrcz) [inline lib/request-wrapper.ts]: Please change this part as stated above. We are not accepting pull requests that would cause breaking changes in our projects.

**Review by diatrcz — CHANGES_REQUESTED** [review]

File: package.json [hunk 3]
```diff
@@ -60,6 +60,7 @@
     "eslint-plugin-jsdoc": "^34.6.3",
     "eslint-plugin-node": "^9.0.0",
     "eslint-plugin-prettier": "^3.0.1",
+    "extend": "3.0.2",
```
Comment (diatrcz) [inline package.json]: As this PR focuses on removing the `extend` dependency from the project, it should also be removed from `devDependencies`.

**Review by diatrcz — CHANGES_REQUESTED** [review]
The changes look good. However, the dependency is still listed in `devDependencies`. To complete the removal of the `extend` dependency, it should also be removed from there, even if this requires additional updates to the test codebase.

**Comment by gameroman** [comment]
Weird, not sure why `npm ci` is failing. I've ran it locally and the lockfile is up-to-date

**Comment by diatrcz** [comment]
If I run it locally `npm ci` fails with the same error message. Running `npm i` adds the missing package to `package-lock.json`.

**Comment by gameroman** [comment]
> If I run it locally `npm ci` fails with the same error message. Running `npm i` adds the missing package to `package-lock.json`.

Would you be able to push that change to this branch please?

I've ran `npm i` and `npm ci` again and there no changes to the lockfile happening locally for me and I am not sure why

Or show the diif that you are getting

**Review by diatrcz — APPROVED** [review]
Looks great, thank you for the contribution!

**Review by pyrooka — APPROVED** [review]
LGTM!
